pkg/codegen/testing/test/testdata/external-enum/nodejs/utilities.ts

```diff
@@ -52,3 +52,17 @@ export function getVersion(): string {
 export function resourceOptsDefaults(): any {
     return { version: getVersion() };
 }
+
+/** @internal */
+export function lazyLoad(exports: any, props: string[], loadModule: any) {
+    let cache = {module: null};
+    for (let property of props) {
+        Object.defineProperty(exports, property, {
+            enumerable: true,
+            get: function() {
+                let m = cache.module || (cache.module = loadModule());
+                return m[property];
```

Comment:
I'm not sure this change is any better than the original syntax (though it doesn't hurt either). NodeJS caches calls to `require()`, so "reloading" the module is free.